src/components/PopupWithForm.js

```diff
@@ -1,11 +1,19 @@
 import Popup from "./Popup.js";
 
 export default class PopupWithForm extends Popup {
-  constructor(popupSelector, handleFormSubmit) {
+  constructor(popupSelector, handleFormSubmit, shouldResetBeforeSubmit = true) {
     super(popupSelector);
     this._popupForm = this._popupElement.querySelector(".modal__form");
     this._handleFormSubmit = handleFormSubmit;
     this._inputList = this._popupElement.querySelectorAll(".modal__input");
+    this._submitButton = this._popupForm.querySelector(".modal__button");
+    this._initialButtonText = this._submitButton.textContent;
+    this._shouldResetBeforeSubmit = shouldResetBeforeSubmit;
+  }
+
+  // Override open to reset form
+  open() {
+    super.open();
   }
```

Comment:
You shouldn’t create the **same methods** of the parent class if you don’t change them inside and only call them with `super`. The parent methods are available in any inheritor of the class.